App.xaml.cs

```diff
@@ -10,13 +11,32 @@ public partial class App : Application
     {
         private void Application_Startup(object sender, StartupEventArgs e)
         {
+            string[] availableLocale = { "en-US", "ja-JP" };
             // mandatory filename arg - grabbed output will write to this file.
             if (e.Args.Length == 1 && !string.IsNullOrEmpty(e.Args[0]))
             {
+                System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("en-US");
                 string filename = e.Args[0];
                 SnippingWindow win = new SnippingWindow(filename);
                 win.Show();
             }
+            else if (e.Args.Length == 2 && !string.IsNullOrEmpty(e.Args[0]) && !string.IsNullOrEmpty(e.Args[1]))
```

Comment:
Rather than making a check here, I would rather pull out all the `args` and figure out what are for what, from their positions.

Reply:
agree, should just extract args and set as appropriate, right now there are three places where setting CultureInfo and two places were creating SnippingWindow.